frontend/src/components/pages/RoutePlanningPage.tsx

```diff
@@ -5,8 +5,19 @@ import { useSearchParams } from "react-router-dom";
 import { useIsPhone } from "@/components/context/PhoneContext.tsx";
 import { createOrUpdateRoute } from "@/api/map/route_store.ts";
 import { Button } from "@/components/ui/button.tsx";
+import type { Point, RouteSegment } from "@/maps/manage/structure.ts";
+import { useEffect } from "react";
 
-const MapWrapper = () => {
+const MapWrapper = ({
```

Comment:
MapWrapper is becoming too fat to still be put in RoutePlanningPage. Can you please migrate the definition of it into MapWrapper.tsx and place it alongside MapView and MapPanel in frontend/src/maps/ folder?